### ISSUE TITLE
Nicer printing of headers/parameters

### ISSUE THREAD

**Issue by schveiguy** [andrewlalis/handy-httpd#28]
**Is your feature request related to a problem? Please describe.**
```d
writeln(context.request.headers);
```

This produces:

```
const(MultiValueMap!(string, string, __lambda6))([const(Entry)("Accept", ["text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8"]), const(Entry)("Accept-Encoding", ["gzip, deflate, br"]), ...
```

**Describe the solution you'd like**
I would like a nicer output, something like:

```
"Accept" : "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8", "Accept-Encoding" : "gzip, deflate, br", ...
```
Where we don't get all the types printed, and single values have their brackets removed.

**Comment by andrewlalis** [comment]
Would you be alright with a multi-line output, like this?
```
"Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8",
"Accept-Encoding": "gzip, deflate, br",
"Accept-Language": "en-US,nl;q=0.7,en;q=0.3",
"Connection": "keep-alive",
"Cookie": "Idea-3fca93ca=273edabd-5ec1-4fb7-9ab3-1e5829a883e8",
"DNT": "1",
"Host": "localhost:8080",
"Sec-Fetch-Dest": "document",
"Sec-Fetch-Mode": "navigate",
"Sec-Fetch-Site": "none",
"Sec-Fetch-User": "?1",
"Upgrade-Insecure-Requests": "1",
"User-Agent": "Mozilla/5.0 (X11; Linux x86_64; rv:121.0) Gecko/20100101 Firefox/121.0"
```
I find it more readable than if everything is dumped onto a single line, especially if you start having multiple values for a header.

**Comment by schveiguy** [comment]
The reason I asked for that format is specifically because that's what AAs do.

Note that with D's formatting facility, it's possible to format a range's elements however you want. I *think* for this whole thing to work, you can just define a `toString` for `Entry`, and this will be used for each element as long as it sees the thing as a range.

One thing I was thinking to suggest is to remove the range functions from `MultiValueMap`, and make an `opIndex()` function that allows fetching just the entry array directly. This way, you can do (e.g. to get the format you asked for):

```d
writefln("%(%s\n%)", context.request.headers[]);
```

Using the range API gives the most flexible interface I think, and makes it possible to do these things. I find the addition of the range functions in `MultiValueMap` to be confusing, especially when it defines things like `opIndex`, but not in a range-way. However, with `context.request.headers` being const, it cannot be considered a range, so I think there is no harm in removing those methods.

So in short, I recommend the range mechanism for flexibility, but also would be fine with an explicit `toString` that uses newlines instead of commas.

**Comment by andrewlalis** [comment]
Okay, that makes sense; I've implemented `toString` for Entry, and now I just use std.format in the `toString` implementation of the MultiValueMap itself. I also went ahead and removed the range methods, and added an opIndex overload for the empty index, which just returns the entries. Will tag the release shortly as v8.1.0.
